presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1327,7 +1330,29 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
                 Optional.empty(),
                 discretePredicates,
                 ImmutableList.of(),
-                EMPTY_STATISTICS);
+                getTableStatistics(hivePartitions));
+    }
+
+    private TableStatistics getTableStatistics(List<HivePartition> hivePartitions)
+    {
+        List<Long> partitionRowNums = hivePartitions.stream()
+                .map(HivePartition::getStatistics)
+                .map(PartitionStatistics::getNumRows)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(toList());
```

Comment:
I need to materialize. Otherwise I would need to use stream twice below.
Once to compute `numRowsSum` and then `partitionsWithStatsCount`.
This is not possible with stream as it is read-once structure.